### ISSUE TITLE
Bug Report: Upscale only updates some tiles

### ISSUE THREAD

**Issue by caretakerscurse** [Coyote-A/ultimate-upscale-for-automatic1111#130]
I'm having a very strange issue with Ultimate SD Upscale. I have tried searching for anyone also having the issue but can't seem to find anything. Maybe my googlefu is just lacking.

Here I just rendered a simple image at 512x using toonyou_beta6 and mostly default settings.![2023-10-27-013039-3369071051](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/60739494/8eaff061-ee89-46aa-adb6-032e8e32044e)

I send this output to image2image, select the USDUpscale script, set scale of 2, Upscaler to R-ESRGAN 4x+ Anime6B (I have tried others, same results). Everything else is unchanged.

For the sake of this example I have the Denoising Strength set to 1 so any changes are very clear. This is what I get out.
![2023-10-27-015354-3369071051](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/60739494/d811bdec-9811-46c9-8de7-535871f2ebd2)

You can see that the first tile is completely changed, as expected. The second block is mostly changed, but the top of this tile is not (this is consistent on every upscale that I have done). The remaining two blocks are just the OG image. No changes other than the edges that touch the first two tiles.

I do not understand what's causing this, nor can I find any other posts reporting the same issue. I've tweaked just about every setting I can think of, but my output always has this same error. Even using Chess as the type, the same areas are unchanged.

Thank you for anyone who took the time to read this. I'm sorry if I've wasted your time here and there is something I am doing wrong. I have included extra information below including sysinfo and the term output.

![Screenshot 2023-10-27 at 02 22 09](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/60739494/9e4e9bf8-c936-44b5-8da9-956b7ee88644)

``` Image size: 512x512
Scale factor: 2
Upscaling iteration 1 with scale factor 2
Tile size: 512x512
Tiles amount: 4
Grid: 2x2
Redraw enabled: True
Seams fix mode: NONE
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [01:13<00:00,  3.68s/it]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [01:13<00:00,  3.68s/it]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [01:13<00:00,  3.69s/it]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [01:13<00:00,  3.69s/it]
Total progress: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 80/80 [04:56<00:00,  3.70s/it]
Total progress: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 80/80 [04:56<00:00,  3.68s/it]```

[sysinfo-2023-10-27-07-36.txt](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/files/13186245/sysinfo-2023-10-27-07-36.txt)

**Comment by caretakerscurse** [comment]
I have now tried to reinstall USD Upscale as well reinstall A1111 completely, save for the models. Still have the same issue.

**Comment by GothWave** [comment]
24/10 is the last time that i used this with perfect results.
![00018-3176302289](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/1fc9b060-adef-4a5f-97a2-72e9de5e7771)

![Screenshot_5](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/a1210908-d45d-4889-bf43-972441b44a44)

![Screenshot_6](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/4368a085-739d-4c25-ac45-c75bc125edf9)

![Screenshot_7](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/2aab11c3-d8c2-45b2-9c25-eae50c42f294)

look how is exactly the same problem, only the first tile + 90% of the entire first line is working
![Screenshot_8](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/9a0a4332-8162-4cb8-b91d-d7164213a34e)
here is 256p in tile width.

![Screenshot_9](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/11e3940d-659d-4f5c-8eb4-2050d332d4df)
and here is with chess in the pattern type (every other image was in Line pattern type)

![Screenshot_10](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/e646ff5b-4526-484b-9a34-914f18447cb0)
well, here is fine... but with pattern type set to None, what means that ultimate upscale did nothing, its just resized.

**Comment by caretakerscurse** [comment]
@GothWave I'm sorry you're also having the issue but I'm relieved to see someone also having the same issue. I felt crazy. 

What hardware are you working with? My setup was an RX7600

**Comment by GothWave** [comment]
My is an RX6600, ryzen 5. The problem is that on 24/10 it was working, so it is something related with a new commit. I will try to run an old version and use --no-half just to make sure (we have AMD, so if i need --no-half i would not be surprise)

**Comment by GothWave** [comment]
![Screenshot_11](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/70006051/60ee1260-5d27-4f1b-bb8b-50458a9117a3)

uhh... its working? i guess...
(using --no-half, with 30% improved performance(?). damn amd)

**Comment by caretakerscurse** [comment]
Yep. That seem to fix it. I hate having `--no-half` on because of all the memory errors, but at least it works. Good find.

Hopefully someone with more know how then us can use this to patch accordingly (if that's even possible).

I'll leave this report open and let the maintainers make the call on what to do about it.

**Comment by CodeingJack** [comment]
My SD have this problem too.
![00001-1099047828](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/142645057/a19236ec-97f5-4f24-8b7a-6b333af63838)
![image](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/assets/142645057/cd68432b-1eec-4600-824b-f8b9f356e2e5)
I use directml. [this](https://github.com/lshqqytiger/stable-diffusion-webui-directml/)
I don't use any other extensions
txt2img has no problem
model is AOM3B2_orangemixs
[sysinfo-2024-01-01-05-52.json](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111/files/13803656/sysinfo-2024-01-01-05-52.json)

**Comment by TheSeventhCode** [comment]
I've been running into this problem to this day, and I'm unsure what exactly causes it.
I'm using sd-ultimate-upscale in combination with control net tile-resample. Sometimes this works, sometimes it doesn't, really weird. I've also added in ADetailer at certain points, which sometimes works as well and sometimes doesn't -.-

I'm retrying now with the `--no-half` argument and see what happens…
